modules/dists/CyclicDist.chpl

```diff
@@ -1296,7 +1296,9 @@ proc CyclicArr.dsiLocalSubdomain(loc: locale) {
 proc CyclicDom.dsiLocalSubdomain(loc: locale) {
   const (gotit, locid) = dist.chpl__locToLocIdx(loc);
   if (gotit) {
-    return whole[(...(chpl__computeCyclic(this.idxType, locid, dist.targetLocDom.dims(), dist.startIdx)))];
+    return whole[(...(chpl__computeCyclic(this.idxType, locid, dist.targetLocDom.dims(), dist.startIdx)))]
+      // supports deprecation by Vass in 1.31 to implement #17131
+      : domain(rank, idxType, stridable=true);
```

Comment:
Why do we need casts like these?

Reply:
The enclosing function has two return statements. The types of their return expressions need to match in order for the function to resolve. This return expression may have a domain type with a more specific value of `strides` than `strideKind.any`, whereas the other one always has `strideKind.any`, translated from `stridable=true`.

So maybe the right solution is to have the other return expression match the type of this return expression. Since I will be updating Cyclic to get rid of `stridable` altogether when `stridable` is deprecated in an upcoming PR, I will do that at that point.